### ISSUE TITLE
Bug VirtualScroll with radio-group

### ISSUE THREAD

**Issue by aspitrine** [ionic-team/ionic#10018]
Ionic version: (check one with "x")
[ ] 1.x
[x] 2.x

I'm submitting a ... (check one with "x")
[x] bug report
[ ] feature request

Current behavior:
I have a list of items with a virtualScroll in a Tabs.
When I select an item, I redirect to the second tabs with radio-group button.
When i select a radio, radio is not selected correctly, the view not refresh data change

Expected behavior:
Select correctly my radio button

Please, I really need this virtualScroll 😩

Steps to reproduce:
Plunkr : http://plnkr.co/edit/cSXG7E

cc #9868

**Comment by brandyscarney** [comment]
@aspitrine Thanks for the issue! Could you run `ionic info` inside of your project and paste the output here please?

**Comment by aspitrine** [comment]
Of course

Cordova CLI: 6.4.0 
Ionic Framework Version: 2.0.0-rc.4
Ionic CLI Version: 2.1.18
Ionic App Lib Version: 2.1.9
Ionic App Scripts Version: 0.0.48
ios-deploy version: 1.9.0 
ios-sim version: 5.0.13 
OS: macOS Sierra
Node Version: v6.9.1
Xcode version: Xcode 8.2.1 Build version 8C1002

**Comment by ghenry22** [comment]
I see very similar with toggles, after navigating from the page with virtualscroll list to the next page the change detection does not work at all on the pushed page.

It's like there is a silent error or something in the background.

I did find that if I do a search on the virtual scroll list first ( even if I then clear it) then navigate to the next page change detection is working as normal.

**Comment by ghenry22** [comment]
Ok I have done some further testing with this and in my case even when the second page is basically blank, no toggles, radios etc and also not using tabs, just a virtualscroll list on the first page, nav.push with some navParams to the second page and change detection is broken on the second page.

IF I do a search on the virtualscroll list (can be done programmatically with a time out, change search term to test after 2000ms, change back to blank after 3000ms) and THEN click on a list item to navigate to the next page everything works as normal.

Really weird, it's almost like virtualScroll stops change detection from being properly initialised on all subsequent pages unless a change is made to the virtualscroll list first.

I did see there were some issues around virtualscroll not working with changeDetectionStrategy.OnPush on github but the symptoms of that do not match these symptoms.

This exact same behaviour occurs on both Android and iOS devices which suggests that it isn't platform or browser specific.  Simply replacing the virtualscroll list on the first page with an ngFor list and everything works perfectly as expected.  Which indicates that the source data used for the array and other elements on the first and second pages are not at fault.  This all points to something with the virtualScroll code itself to me.

@jgw96 & @brandyscarney - Any help on this would be really appreciated.  I can reproduce the problem consistently and am happy to test any suggested changes or nightlies.

**Comment by zakton5** [comment]
+1 Virtual scroll continues to break in new ways after each iteration. This really needs to be fixed as soon as possible :)

**Comment by MattDHill** [comment]
I'm experiencing a similar issue with checkboxes (selecting multiple entries from a long list of contacts). Upon first entering the view, everything seems to work OK. But, when I destroy the view and then re-enter, the the virtual list will sometimes fail to display the selected attribute (i.e. the boxes don't display a check, even though the handler fires). Then, if I set focus on the search bar (thus opening the keyboard) the list redraws and all the checkmarks suddenly appear. This behavior is very random and unpredictable, I have not been able to consistently reproduce with a specific set of clicks. But it's bad enough that virtual scroll is currently unusable for a long list of checkbox items. I've tried many configurations with approxItemHeight, excluding the searchbar, header sizes, even keyboard settings, etc etc. Nothing has worked and there are no errors of any kind.
